api/tests/integration/infrastructure/repositories/certification-center-membership-repository_test.js

```diff
@@ -509,4 +509,89 @@ describe('Integration | Repository | Certification Center Membership', function
       });
     });
   });
+
+  describe('#disableMembershipsByUserId', function () {
+    const now = new Date('2022-12-05');
+    let clock;
+
+    beforeEach(function () {
+      clock = sinon.useFakeTimers(now);
+    });
+
+    afterEach(function () {
+      clock.restore();
+    });
+
+    context('when there is multiple memberships for the specified user id', function () {
+      it('disabled all memberships of this user', async function () {
+        // given
+        const userId = databaseBuilder.factory.buildUser().id;
+        const updatedByUserId = databaseBuilder.factory.buildUser().id;
+        const firstCertificationCenterId = databaseBuilder.factory.buildCertificationCenter().id;
+        const secondCertificationCenterId = databaseBuilder.factory.buildCertificationCenter().id;
+        const firstMembership = databaseBuilder.factory.buildCertificationCenterMembership({
+          userId,
+          certificationCenterId: firstCertificationCenterId,
+          createdAt: now,
+        });
+        const secondMembership = databaseBuilder.factory.buildCertificationCenterMembership({
+          userId,
+          certificationCenterId: secondCertificationCenterId,
+          createdAt: now,
+        });
+        databaseBuilder.factory.buildCertificationCenterMembership({
+          userId: databaseBuilder.factory.buildUser().id,
+          certificationCenterId: secondCertificationCenterId,
+          createdAt: now,
+        });
+
+        await databaseBuilder.commit();
+
+        const expectedMemberships = [
+          {
+            ...firstMembership,
+            createdAt: now,
+            disabledAt: now,
+            updatedByUserId,
+          },
+          {
+            ...secondMembership,
+            createdAt: now,
+            disabledAt: now,
+            updatedByUserId,
+          },
+        ];
+
+        // when
+        await certificationCenterMembershipRepository.disableMembershipsByUserId({ userId, updatedByUserId });
+
+        // then
+        const disabledMemberships = await knex('certification-center-memberships').returning('*').where({ userId });
+        expect(disabledMemberships.length).to.equal(2);
+        expect(disabledMemberships).to.deep.include.members(expectedMemberships);
+      });
+    });
+
+    context('when there is no membership for the specified user id', function () {
```

Comment:
je suis partagé pour ce test, on ne vérifie pas grand chose en vérifiant ça `expect(disabledMemberships.length).to.equal(0);` 

A la rigueur on peut s'assurer qu'il ne disable pas des memberships qui ne lui appartient pas

Reply:
Si la requête n'était pas correcte alors on aurait reçu des éléments. Des éléments sont retournés uniquement si une modification en base de données a été faite.

Reply:
> Si la requête n'était pas correcte alors on aurait reçu des éléments

Tu veux dire que tu testes ici que l'utilisateur ne se retrouve pas soudainement avec un membership alors qu'il n'en avait pas à la base ?

Sauf si je loupe un truc mais en voyant le code : pas de membership dans le given et la requête du then cherche les memberships de l'utilisateur, du coup peut importe comment tu écris ta requête de disable, tu ne recevra jamais d'élément. 🤔
Sauf si tu écris une requête de création de membership bien sûr 😅

Reply:
> Tu veux dire que tu testes ici que l'utilisateur ne se retrouve pas soudainement avec un membership alors qu'il n'en avait pas à la base ?

`.update()` de knex va retourner les éléments si modification il y a eu, donc si il a trouvé des éléments à modifier.
- Si l'utilisateur n'est pas membre, knex va retourner un tableau vide (aucune action n'a été faite).
- Si l'utilisateur est membre d'un ou plusieurs centres de certification, knex va retourner le tableau d'éléments qui a été modifié.

Reply:
Je viens de comprendre ta remarque 💡. J'ai changé le test et ajouté un autre test.